File: LeetCode/101. Symmetric Tree/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        stackL = []
        stackR = []

        stackL.append(root.left)
        stackR.append(root.right)
        
        while stackL:
            l = stackL.pop()
            r = stackR.pop()

            if not l and not r: continue
            if not l: return False
            if not r: return False

            if l.val != r.val: return False

            stackL.append(l.left)
            stackL.append(l.right)

            stackR.append(r.right)
            stackR.append(r.left)

        return True

File: LeetCode/101. Symmetric Tree/Solution.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution
{
    public bool IsSymmetric(TreeNode root)
    {
        var stackL = new Stack<TreeNode>();
        var stackR = new Stack<TreeNode>();

        stackL.Push(root.left);
        stackR.Push(root.right);

        while (stackL.Count() != 0)
        {
            var l = stackL.Pop();
            var r = stackR.Pop();

            if (l == null && r == null) continue;
            if (l == null) return false;
            if (r == null) return false;

            if (l.val != r.val) return false;

            stackL.Push(l.left);
            stackL.Push(l.right);

            stackR.Push(r.right);
            stackR.Push(r.left);
        }

        return true;
    }
}